src/platform_impl/web/window.rs

```diff
@@ -312,7 +312,7 @@ impl Inner {
     #[inline]
     pub(crate) fn fullscreen(&self) -> Option<Fullscreen> {
         if self.canvas.is_fullscreen() {
-            Some(Fullscreen::Borderless(None))
+            Some(Fullscreen::Borderless(Some(self.monitor.current_monitor())))
```

Comment:
Not sure if this would necessarily be better? Isn't the "correct" thing to do here to track which state the user originally requested to fullscreen in (custom monitor chosen, or no monitor chosen), and then return the same here? (Such that the assertion in the sequence `window.set_fullscreen(fullscreen); assert_eq!(window.fullscreen(), fullscreen)` would hold).

Reply:
I was unsure as well how this would play out, but after some testing I've learned that the current screen is always the one requested to be fullscreen, regardless of what it was before.

So yes, your assertion would hold.

Reply:
I mean, if I set `let fullscreen = Fullscreen::Borderless(None)`, the assertion no longer holds.

Reply:
Ah, I see!
No, in that case it won't.

I'm not entirely sure what we should do here exactly, but I'm actually in favor of the current way Web works: always return the currently correct values, regardless of user input.
E.g. what do I do if `set_fullscreen()` fails? Should the assert hold until I receive the error?

This is all less then ideal and should be documented thoroughly ... so I'm gonna leave a comment in https://github.com/rust-windowing/winit/issues/3690.